hooks/workfiles_management.py

```diff
@@ -41,12 +42,17 @@ class WorkfilesManagement(sgtk.get_hook_baseclass()):
     # sg_path_cache_storage (local storage entity): LocalStorage the file is part of. This can't
     #   be created through the GUI, but can be programmatically:
     #   shotgun.schema_field_create("CustomEntity45", "entity", "Path Cache Storage", {"valid_types": ["LocalStorage"]})
+    #
+    # Note that the code field in Shotgun will be the name of the file on disk.
+    # The sg_name_field value however is the value of the {name} token inside the file name,
+    # e.g. given the template {name}.v{version}.ma and the file name (code) of car.v003.ma,
+    # sg_name_field would be set to "car".
 
     def register_workfile(self, name, version, context, work_template, path, description, image):
         """
         Register a work file with Shotgun.
 
-        :param name str: Name of the work file.
+        :param name str: File name on disk of the work file. Not to be confused with the {name} token.
```

Comment:
@jfboismenu why not rename the `name` parameter to be `filename` for clarity? If you are leaving it for compatibility reasons, let's discuss! There aren't yet that many users of this (and the main user submitted this PR hehe). If you chose name for symmetry reasons to match other code, it's a different discussion and then we probably shouldn't mess with renaming it.... :)